File: programmers/python/programmers_stackqueue_1.py
import math

def solution(progresses, speeds):
	answer = []
	completion = []
	for i in range(len(progresses)):
		completion.append(math.ceil((100 - progresses[i]) / speeds[i]))
	val = completion[0]
	answer.append(1)
	idx = 0
	for i in range(1, len(completion)):
		if completion[i] <= val:
			answer[idx] += 1
		else:
			val = completion[i]
			idx += 1
			answer.append(1)
	return answer

File: programmers/cpp/programmers_stackqueue_1.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;

	while (1)
	{
		if (progresses.empty())
			break;
		int cnt = 0;
		for (int i = 0; i < progresses.size(); i++)
			progresses[i] += speeds[i];
		for (int i = 0; i < progresses.size(); i++)
		{
			if (progresses[i] < 100)
				break;
			cnt++;
		}
		if (cnt != 0)
		{
			progresses.erase(progresses.begin(), progresses.begin() + cnt);
			speeds.erase(speeds.begin(), speeds.begin() + cnt);
			answer.push_back(cnt);
			cnt = 0;
		}
	}

	return answer;
}